src/utils/is-gas-environment.ts

```diff
@@ -0,0 +1,3 @@
+const isGASEnvironment = (): boolean => typeof google !== 'undefined' && Boolean(google?.script?.run);
```

Comment:
By the way, one upside to using webpack to set up environment variables is that webpack can know at build time whether to use the proxy (dev) or the promisified (prod) version, and only include that code. With the new approach, all the proxy and iframe event code is getting bundled in both dev and prod builds since it only knows at runtime which environment it's in. Any thoughts on how to remove the unneeded code in prod builds?

Reply:
I left this as is in #19. There was actually an [issue with Safari](https://github.com/enuchi/React-Google-Apps-Script/issues/94) with the way I was doing it based on the error message thrown (Safari had a different error message :grin:) so the way you did it works better.